pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -61,6 +61,23 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
 	return nil
 }
 
+func (r *ReconcileStorageCluster) newOBCStorageClass(initData *ocsv1.StorageCluster) *storagev1.StorageClass {
+	reclaimPolicy := corev1.PersistentVolumeReclaimDelete
+	retSC := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-ceph-%s-bucket", initData.Name, reclaimPolicy),
```

Comment:
I'm not sure if I missed this earlier, but I don't like this naming scheme. First, you should create a `generateNameForCephRgwSC()` function to hold the name generation. Second, the name should follow the pattern of the other StorageClasses, just `<name>-ceph-rgw`. Since we're only ever creating one of these right now, there's no need to specify the reclaimPolicy in the name.

Reply:
Added `generateNameForCephRgwSC()` function
Added tests to cover all the platforms.